onadata/libs/utils/middleware.py

```diff
@@ -64,3 +75,20 @@ def __call__(self, request):
                     query['time'], " ".join(query['sql'].split())))
 
         return response
+
+
+class OperationalErrorMiddleware(BaseMiddleware):
+    """
+    Captures requests returning 500 status code.
+    Then retry it against master database.
+    """
+    def process_exception(self, request, exception):
+        # Filter out OperationalError Exceptions
+        if isinstance(exception, OperationalError):
+            already_raised = getattr(settings, "ALREADY_RAISED", False)
+            if not already_raised:
+                settings.ALREADY_RAISED = True
+                with use_master:
+                    response = self.get_response(request)
+                    return response
+            settings.ALREADY_RAISED = None
```

Comment:
I'd advise keeping this value as a bool and hence have either `True` or`False` as its value and not `None`.

Reply:
I am okay with `False`.